Tasks/Src/Task_FaultState.c

```diff
@@ -145,6 +142,8 @@ void EnterFaultState() {
 #ifdef DEBUGMODE
     char command[COMMAND_SIZE];
 #endif
+    BSP_WDTimer_Init(); //This is in case we did not pass critical state and watchdog timer was not initialized
+    BSP_WDTimer_Start(); 
```

Comment:
Maybe we just want to not use the watchdog timer if this is the case because we don't want it to kick us out of the fault state?

Reply:
This is for actual BPS, not just the simulator. If we do not pass the critical state, then the fault state breaks because it tries to reset a timer that wasn't initialized. But if we do pass the critical state and then fault, then we need to reset the watchdog timer.

Reply:
Yes. I'm suggesting that actual BPS should only try and reset the watchdog timer if the timer has been initialized already, since there is no point in running the watchdog timer if we are already in a fault state

Reply:
I see, I will add an issue ticket for that.